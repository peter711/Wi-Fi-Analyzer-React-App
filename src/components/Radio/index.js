import React from 'react';
import { RadioWrapper, RadioInputWrapper } from './styles';

class Radio extends React.PureComponent {
  render() {
    const { label, options } = this.props;
    return (
      <RadioWrapper>
        <label>{label}</label>
        {options && renderRadios(this.props)}
      </RadioWrapper>
    );
  }
}

export default Radio;

/////////////////////////////////////////////////////////////

function renderRadios(props) {
  const { onChange, value } = props;
  return (
    props.options.map((option, index) => (
      renderRadio(option, index, { onChange, currentValue: value })
    ))
  );
}

function renderRadio({ text, value }, index, { onChange, currentValue }) {
  return (
    <RadioInputWrapper key={index}>
      <label>
        {text}
        <input
          type="radio"
          name="radioInput"
          value={value}
          checked={value === currentValue}
          onChange={(event) => {
            const value = event.target.value;
            onChange(value);
          }}
        />
        <span className="checkmark" />
      </label>
    </RadioInputWrapper>
  );
}
