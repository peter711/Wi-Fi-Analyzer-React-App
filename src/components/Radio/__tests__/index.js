import React from "react";
import { shallow } from "enzyme";

import Radio from "../";

const options = [
    { value: 'foo', text: 'Foo' },
    { value: 'bar', text: 'Bar' }
];

describe("<Radio/>", () => {
    it("should render", () => {
        const component = shallow(<Radio />);
        expect(component).toMatchSnapshot();
    });

    it("should render radio options", () => {
        const component = shallow(<Radio options={options} />);
        expect(component.find('input')).toHaveLength(2);
        expect(component).toMatchSnapshot();
    });

    it("should call onChange", () => {
        const value = 'bar';
        const onChange = jest.fn();
        const component = shallow(<Radio onChange={onChange} options={options} />);

        expect(component).toMatchSnapshot();
        component.find('input').at(1).simulate('change', {
            target: { value }
        });

        expect(onChange).toBeCalledWith(value);
    });
});