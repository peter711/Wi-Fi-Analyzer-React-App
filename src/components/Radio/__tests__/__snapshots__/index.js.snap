// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio/> should call onChange 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Radio
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "text": "Foo",
          "value": "foo",
        },
        Object {
          "text": "Bar",
          "value": "bar",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label />,
        Array [
          <styled.div>
            <label>
              Foo
              <input
                checked={false}
                name="radioInput"
                onChange={[Function]}
                type="radio"
                value="foo"
              />
              <span
                className="checkmark"
              />
            </label>
          </styled.div>,
          <styled.div>
            <label>
              Bar
              <input
                checked={false}
                name="radioInput"
                onChange={[Function]}
                type="radio"
                value="bar"
              />
              <span
                className="checkmark"
              />
            </label>
          </styled.div>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": <label>
            Foo
            <input
              checked={false}
              name="radioInput"
              onChange={[Function]}
              type="radio"
              value="foo"
            />
            <span
              className="checkmark"
            />
          </label>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Foo",
              <input
                checked={false}
                name="radioInput"
                onChange={[Function]}
                type="radio"
                value="foo"
              />,
              <span
                className="checkmark"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Foo",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "name": "radioInput",
                "onChange": [Function],
                "type": "radio",
                "value": "foo",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "checkmark",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "label",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": <label>
            Bar
            <input
              checked={false}
              name="radioInput"
              onChange={[Function]}
              type="radio"
              value="bar"
            />
            <span
              className="checkmark"
            />
          </label>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Bar",
              <input
                checked={false}
                name="radioInput"
                onChange={[Function]}
                type="radio"
                value="bar"
              />,
              <span
                className="checkmark"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Bar",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "name": "radioInput",
                "onChange": [Function],
                "type": "radio",
                "value": "bar",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "checkmark",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "label",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label />,
          Array [
            <styled.div>
              <label>
                Foo
                <input
                  checked={false}
                  name="radioInput"
                  onChange={[Function]}
                  type="radio"
                  value="foo"
                />
                <span
                  className="checkmark"
                />
              </label>
            </styled.div>,
            <styled.div>
              <label>
                Bar
                <input
                  checked={false}
                  name="radioInput"
                  onChange={[Function]}
                  type="radio"
                  value="bar"
                />
                <span
                  className="checkmark"
                />
              </label>
            </styled.div>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": <label>
              Foo
              <input
                checked={false}
                name="radioInput"
                onChange={[Function]}
                type="radio"
                value="foo"
              />
              <span
                className="checkmark"
              />
            </label>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Foo",
                <input
                  checked={false}
                  name="radioInput"
                  onChange={[Function]}
                  type="radio"
                  value="foo"
                />,
                <span
                  className="checkmark"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Foo",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "name": "radioInput",
                  "onChange": [Function],
                  "type": "radio",
                  "value": "foo",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "checkmark",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": <label>
              Bar
              <input
                checked={false}
                name="radioInput"
                onChange={[Function]}
                type="radio"
                value="bar"
              />
              <span
                className="checkmark"
              />
            </label>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Bar",
                <input
                  checked={false}
                  name="radioInput"
                  onChange={[Function]}
                  type="radio"
                  value="bar"
                />,
                <span
                  className="checkmark"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Bar",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "name": "radioInput",
                  "onChange": [Function],
                  "type": "radio",
                  "value": "bar",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "checkmark",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Radio/> should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Radio />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label />,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label />,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Radio/> should render radio options 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Radio
    options={
      Array [
        Object {
          "text": "Foo",
          "value": "foo",
        },
        Object {
          "text": "Bar",
          "value": "bar",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label />,
        Array [
          <styled.div>
            <label>
              Foo
              <input
                checked={false}
                name="radioInput"
                onChange={[Function]}
                type="radio"
                value="foo"
              />
              <span
                className="checkmark"
              />
            </label>
          </styled.div>,
          <styled.div>
            <label>
              Bar
              <input
                checked={false}
                name="radioInput"
                onChange={[Function]}
                type="radio"
                value="bar"
              />
              <span
                className="checkmark"
              />
            </label>
          </styled.div>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": <label>
            Foo
            <input
              checked={false}
              name="radioInput"
              onChange={[Function]}
              type="radio"
              value="foo"
            />
            <span
              className="checkmark"
            />
          </label>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Foo",
              <input
                checked={false}
                name="radioInput"
                onChange={[Function]}
                type="radio"
                value="foo"
              />,
              <span
                className="checkmark"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Foo",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "name": "radioInput",
                "onChange": [Function],
                "type": "radio",
                "value": "foo",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "checkmark",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "label",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": <label>
            Bar
            <input
              checked={false}
              name="radioInput"
              onChange={[Function]}
              type="radio"
              value="bar"
            />
            <span
              className="checkmark"
            />
          </label>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Bar",
              <input
                checked={false}
                name="radioInput"
                onChange={[Function]}
                type="radio"
                value="bar"
              />,
              <span
                className="checkmark"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Bar",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "name": "radioInput",
                "onChange": [Function],
                "type": "radio",
                "value": "bar",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "checkmark",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "label",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label />,
          Array [
            <styled.div>
              <label>
                Foo
                <input
                  checked={false}
                  name="radioInput"
                  onChange={[Function]}
                  type="radio"
                  value="foo"
                />
                <span
                  className="checkmark"
                />
              </label>
            </styled.div>,
            <styled.div>
              <label>
                Bar
                <input
                  checked={false}
                  name="radioInput"
                  onChange={[Function]}
                  type="radio"
                  value="bar"
                />
                <span
                  className="checkmark"
                />
              </label>
            </styled.div>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": <label>
              Foo
              <input
                checked={false}
                name="radioInput"
                onChange={[Function]}
                type="radio"
                value="foo"
              />
              <span
                className="checkmark"
              />
            </label>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Foo",
                <input
                  checked={false}
                  name="radioInput"
                  onChange={[Function]}
                  type="radio"
                  value="foo"
                />,
                <span
                  className="checkmark"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Foo",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "name": "radioInput",
                  "onChange": [Function],
                  "type": "radio",
                  "value": "foo",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "checkmark",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": <label>
              Bar
              <input
                checked={false}
                name="radioInput"
                onChange={[Function]}
                type="radio"
                value="bar"
              />
              <span
                className="checkmark"
              />
            </label>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Bar",
                <input
                  checked={false}
                  name="radioInput"
                  onChange={[Function]}
                  type="radio"
                  value="bar"
                />,
                <span
                  className="checkmark"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Bar",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "name": "radioInput",
                  "onChange": [Function],
                  "type": "radio",
                  "value": "bar",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "checkmark",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
